docs/targets/polkadot.rst

```diff
@@ -62,3 +61,45 @@ The following example shows how call flags can be used:
 .. include:: ../examples/polkadot/call_flags.sol
   :code: solidity
 
+
+Reverts and error data decoding
+_______________________________
+
+When a contract reverts, the returned error data is what
+`EVM would return <https://docs.soliditylang.org/en/v0.8.20/control-structures.html#panic-via-assert-and-error-via-require>`_.
+``assert()``, ``require()``, or ``revert()`` will revert the contract execution, where the revert reason 
+is supplied as the contracts output (if any). Solidity contracts can also revert with a `Panic` 
+(please refer to the 
+`Solidity language documentation <https://docs.soliditylang.org/en/v0.8.20/control-structures.html#panic-via-assert-and-error-via-require>`_.
+for more information about when `Panic` might be returned).
+Uncaught exceptions from calling and instantiating contracts or transferring funds will be bubbled 
+up back to the caller.
+
+The metadata contains all error variants that the contract knows about in the ``lang_error`` field.
+The 4 bytes `selector` of the error data can be seen as the enum discriminator. However, because 
+SCALE encoding does not allow discriminators larger than 1 byte, the hex-encoded error selector 
+is provided as the enum variant name in the metadata (the selector could also be calculated by
+reconstructing and hashing the error signature based on the enum variant types).
```

Comment:
So where does the name go? E.g. `Panic`, `Error`, or `CustomError5`? 

I do think this this paragraph is inscrutable. I still don't have a good idea of what is going on here.

Reply:
Makes me think, do you think it'd be better if the Selector goes into the path of the Variant Type?